Source/Scene/Camera.js

```diff
@@ -3230,9 +3241,39 @@ define([
         Matrix4.clone(camera._transform, result.transform);
         result._transformChanged = true;
 
+        if (defined(camera._boundingObject)) {
+            result.boundingObject = camera._boundingObject.clone(result.boundingObject);
+        }
+
         return result;
     };
 
+    var scratchViewOptions = {
+        destination : new Cartesian3(),
+        orientation : new HeadingPitchRoll()
+    };
+    var scratchClampedValue = new Cartesian3();
+
+    /**
+     * @private
+     */
+    Camera.prototype._limitPosition = function() {
+        if (!defined(this.boundingObject)) {
+            return;
+        }
+
+        scratchViewOptions.orientation.heading = this.heading;
+        scratchViewOptions.orientation.pitch = this.pitch;
+        scratchViewOptions.orientation.roll = this.roll;
+
+        // solving for positionWC because there's a bug where WC is incorrect in 2D and CV mode
+        scratchViewOptions.destination = this.boundingObject.clampToBounds(Cartographic.toCartesian(this._positionCartographic), scratchClampedValue);
```

Comment:
Using `Cartographic.toCartesian` make sit generate a `new Cartesian3()` every time this is called.
This is also not compatible with different ellipsoid types.  Use `this._scene.globe.ellipsoid.cartographicToCartesian` instead.

Also, at the top of the function you'll need to return if `scene.globe` is undefined